components/mpas-framework/Makefile

```diff
@@ -515,6 +515,13 @@ ifeq "$(USE_PIO2)" "true"
 	override CPPFLAGS += -DUSE_PIO2
 endif
 
+#
+# If user has indicated ADIOS isn't supported, define NO_ADIOS pre-processor macro
+#
+ifeq "$(ADIOS)" "false"
+	override CPPFLAGS += -DNO_ADIOS
+endif
+
```

Comment:
I think its better to have cpp flags follow "defined = on = true = yes" logic.  So if ADIOS is requested, set "HAVE_ADIOS" to "True".

Reply:
@rljacob, yes I think that makes sense.  I'll change it.

Reply:
Looks like "USE_ADIOS" would fit the pattern with other cpp constants.

Reply:
@rljacob, it's becoming clear to me why I did it this way.  If the CPP flag is `NO_ADIOS`, then the E3SM default of not defining this flag means ADIOS is supported.  If you prefer the `HAVE_ADIOS` flag, we presumably need to define that always for E3SM.  I was trying to modify only the standalone build infrastructure but I can figure out how to add this to the E3SM build, too.

Reply:
Oh you're right.  Ok to leave it as-is then.

Reply:
I did implement `HAVE_ADIOS`.  It's not super hard to have this flag for E3SM runs.  Shall I still go back to what I had?  I'm fine either way.

Reply:
No keep what you just did.

Reply:
Should line 521 be "true" instead of "false"?

Reply:
No, I think what I have is right.  I think we want the default even for standalone MPAS to be that `HAVE_ADIOS=true`, so you have to explicitly set `HAVE_ADIOS=false` to indicate you're using PIO and don't have ADIOS support.

Reply:
How would we make the default `HAVE_ADIOS=true` if `HAVE_ADIOS` is undefined?

Reply:
The HAVE_ADIOS in `ifeq "$(HAVE_ADIOS)" "true"` is a makefile variable that should be given a default value of false.  If its true, the cpp variable HAVE_ADIOS will be defined.

Reply:
@rljacob, I don't think that's the behavior we want.  We want `HAVE_ADIOS=true` , not `=false`, to be the default.